### PR TITLE
docs(context-outlet-typed): update content

### PR DESCRIPTION
**Pull request by svenson95**
After solving this challenge i've updated the docs a little bit. Added a link to your article in the docs, even it's declared below, because I didn't noticed the article button below - only when I finished the task 😄

### CODE REVIEW

File: docs/src/content/docs/challenges/angular/4-context-outlet-typed.md [hunk 1]
```diff
@@ -16,30 +16,34 @@ sidebar:
```
Comment (svenson95) [inline docs/src/content/docs/challenges/angular/4-context-outlet-typed.md:16]: @tomalaforge can i refactor the command in line 11 (expand this file here in github) to `angular-typed-context-outlet`? So it matches with the name of this challenge - Typed ContextOutlet.

File: docs/src/content/docs/challenges/angular/4-context-outlet-typed.md [hunk 2]
```diff
@@ -16,30 +16,34 @@ sidebar:
 
 ## Information
 
-Angular offer the static function [`ngTemplateContextGuard`](https://angular.io/guide/structural-directives#typing-the-directives-context) to strongly type structural directive.
+You can improve template type checking for custom directives by adding template guard properties to your directive definition. Angular offer the static function [`ngTemplateContextGuard`](https://angular.io/guide/structural-directives#typing-the-directives-context) to strongly type structural directives.
```
Comment (jdegand) [inline docs/src/content/docs/challenges/angular/4-context-outlet-typed.md:19]: offers

File: docs/src/content/docs/challenges/angular/4-context-outlet-typed.md [hunk 3]
```diff
@@ -16,30 +16,34 @@ sidebar:
...
 
 However the context of **NgTemplateOutlet** type is **Object**. But with the help of the above guard, we can improve that behavior.
 
+If you’re not very familiar with `ngTemplateOutlet`, check out this article.
+
+- [NgTemplateOutlet Type Checking](https://medium.com/@thomas.laforge/ngtemplateoutlet-type-checking-5d2dcb07a2c6)
```
Comment (jdegand) [inline docs/src/content/docs/challenges/angular/4-context-outlet-typed.md:25]: You don't need this here, as it is already linked in the article button.

File: docs/src/content/docs/challenges/angular/4-context-outlet-typed.md [hunk 4]
```diff
@@ -16,30 +16,34 @@ sidebar:
...
+
 ## Statement
 
-In this exercise, we want to learn how to strongly type our ng-template in our AppComponent.
+In this exercise we want to learn how to strongly type our `ng-template` in our `AppComponent`.
```
Comment (jdegand) [inline docs/src/content/docs/challenges/angular/4-context-outlet-typed.md:29]: add the comma back

File: docs/src/content/docs/challenges/angular/4-context-outlet-typed.md [hunk 5]
```diff
@@ -16,30 +16,34 @@ sidebar:
...
 
-### Level 2: generic Interface
+### Level 2: Generic Interface
 
 Currently we have the following piece of code.
 
 ![Unkown Student](../../../../assets/4/unknown-student.png 'Unkown Student')
 
-As we can see, student is of type "any". We want to infer the correct type.
+As we can see `student` is of type `any`. We want to infer the correct type using a custom directive `ListDirective`.
```
Comment (jdegand) [inline docs/src/content/docs/challenges/angular/4-context-outlet-typed.md:47]: As we can see,'student'` is of type `any`. We want to infer the correct type using the custom directive `ListDirective`.

Need comma after `see` and change `a` to `the`.

**Review by jdegand — CHANGES_REQUESTED** [review]

File: docs/src/content/docs/challenges/angular/4-context-outlet-typed.md (hunk 3 above)
Comment (svenson95) [reply docs/src/content/docs/challenges/angular/4-context-outlet-typed.md:25]: Check the [PR description](https://github.com/tomalaforge/angular-challenges/pull/705#issue-2200612076), i found this article link after i finished the challenge. And in other docs, e. g. [Structural Directive](https://angular-challenges.vercel.app/challenges/angular/6-permissions/) there are links in the docs too.

File: docs/src/content/docs/challenges/angular/4-context-outlet-typed.md (hunk 2 above)
Comment (svenson95) [reply docs/src/content/docs/challenges/angular/4-context-outlet-typed.md:19]: i need to improve my english 😆

File: docs/src/content/docs/challenges/angular/4-context-outlet-typed.md (hunk 3 above)
Comment (jdegand) [reply docs/src/content/docs/challenges/angular/4-context-outlet-typed.md:25]: Yes, you aren't supposed to read the solution before you try the challenge.  Not all challenges have articles and the articles can help people when there are stuck or lost.     

It is better to add a note in another section about articles i.e. if you get stuck or can't complete the challenge, read a detailed solution by clicking the article button.

File: docs/src/content/docs/challenges/angular/4-context-outlet-typed.md (hunk 2 above)
Comment (jdegand) [reply docs/src/content/docs/challenges/angular/4-context-outlet-typed.md:19]: There are plenty of grammar checkers and translators out there.  Check out [deepl](https://www.deepl.com/en/translator) and [quillbot](https://quillbot.com/).

File: docs/src/content/docs/challenges/angular/4-context-outlet-typed.md (hunk 3 above)
Comment (svenson95) [reply docs/src/content/docs/challenges/angular/4-context-outlet-typed.md:25]: Okay i'll remove the link, nevertheless imo your arguments doesn't make much sense to me. It's upon you if you take the challenge serious and don't look for a solution right away. The community solutions are available any way, for me a challenge itself is about know how to find those informations. And thus find a solution for a problem.

As i was doing this challenge I've also used google to find informations about this topic, and found thomas article about it. But anyway, your point is valid too, we don't need to make it too easy to find the answers.

Comment (jdegand) [reply docs/src/content/docs/challenges/angular/4-context-outlet-typed.md:25]: Well, googling and now prompting are part of the challenge and life of development.  The best challenges and topics are when there is incomplete and poor documentation out there.  This repo can fill the gap.

Comment (svenson95) [reply docs/src/content/docs/challenges/angular/4-context-outlet-typed.md:25]: well said 🚀

**Comment by svenson95** [comment]
@jdegand changes done

**Review by jdegand — APPROVED** [review]

File: docs/src/content/docs/challenges/angular/4-context-outlet-typed.md (hunk 1 above)
Comment (tomalaforge) [reply docs/src/content/docs/challenges/angular/4-context-outlet-typed.md:16]: Yes you can if you want. Do it on a separate PR, It will be easier 👍

**Review by tomalaforge — APPROVED** [review]

**Pull request merged**